pkg/controller/core.oam.dev/v1alpha2/application/apply.go

```diff
@@ -93,108 +69,53 @@ func (h *appHandler) handleErr(err error) (ctrl.Result, error) {
 	}, nil
 }
 
-func (h *appHandler) apply(ctx context.Context, appRev *v1beta1.ApplicationRevision, ac *v1alpha2.ApplicationConfiguration, comps []*v1alpha2.Component, policies []*unstructured.Unstructured) error {
-
-	if h.app.Spec.Workflow != nil || ac.Annotations[oam.AnnotationAppRevisionOnly] == "true" {
-		h.FinalizeAppRevision(appRev, ac, comps)
-		err := h.createOrUpdateAppRevision(ctx, appRev)
-		if err != nil {
-			return err
-		}
-		return h.createResourcesConfigMap(ctx, appRev, ac, comps, policies)
+func (h *appHandler) applyAppManifests(ctx context.Context, comps []*types.ComponentManifest, policies []*unstructured.Unstructured) error {
+	appRev := h.currentAppRev
+	if h.app.Spec.Workflow != nil || appRev.Annotations[oam.AnnotationAppRevisionOnly] == "true" {
```

Comment:
why the appRev has this annotation? I noticed it's ApplicationConfiguration before change.

Reply:
That's true.  Before change, we have to pass application's labels&annotations to AppConfig, After change, I pass these info to AppRevision.